lib/farscape/transition.rb

```diff
@@ -11,24 +12,36 @@ def initialize(transition, agent)
       @transition = transition
     end
 
-    def invoke
-      options = OpenStruct.new
-      yield options if block_given?
+    def invoke(args={})
+      opts=OpenStruct.new
+      yield opts if block_given?
+      options = match_params(args, opts)
 
       call_options = {}
-      call_options[:url] = uri
-      call_options[:method] = interface_method
+      call_options[:url] = @transition.uri
+      call_options[:method] = @transition.interface_method
       call_options[:headers] = @agent.get_accept_header(@agent.media_type).merge(options.headers || {})
       call_options[:params] = options.parameters if options.parameters
       call_options[:body] = options.attributes if options.attributes
 
       response = @agent.client.invoke(call_options)
-      Representor.new(@agent.media_type, response.body || EMPTY_BODIES[@agent.media_type], @agent)
+      @agent.representor.new(@agent.media_type, response.body || EMPTY_BODIES[@agent.media_type], @agent)
     end
-    
-    # TODO: Remove Constants from Representor Classes
+
     def method_missing(meth, *args, &block)
       @transition.send(meth, *args, &block)
     end
+
+    private
+
+    def match_params(args, options)
```

Comment:
If you're grossed out by the repeated `public_send`s, I can't really blame you. The point for me was really to inline the lambdas, since they're now only called once:

```ruby
def match_params(args, options)
      [:parameters, :attributes].each do |key_type|
        field_names = @transition.public_send(key_type).map { |field| field.name.to_sym }
        filtered_values = args.select { |k,_| field_names.include?(k) } }
        options.public_send(:"#{key_type}=", filtered_values.merge(options.public_send(key_type) || {}))
      end
      options
    end
```